async/amqp/integration_test.go

```diff
@@ -0,0 +1,192 @@
+// +build integration
+
+package amqp
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/beatlabs/patron/encoding/json"
+	"github.com/streadway/amqp"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestConsumeAndPublish(t *testing.T) {
+	// Setup consumer.
```

Comment:
the timeouts in these tests are enormous which makes them slow and slow tests are tests not run by developers.

Reply:
I cut it down to 1 second for the initial setup, plus 0.1 seconds for each message. It's mainly a precautionary measure to avoid random flaky fails down the line. What do you think?

Reply:
the faster the better... if tou can get even lower that would be also nice...

Reply:
In my first local tests, lowering these values impacted the flakiness. Right now, the delay is ~1.2 seconds, which is a good compromise IMO. 

I can run some more local and GH Actions tests to see how low we can go, without impacting the passing of valid tests.

Reply:
no worries, let's make it 2s to ad a buffer and leave it there. Seems fast enough for integration tests.